src/modules/shared/components/MultipleSelect/MultipleSelect.tsx

```diff
@@ -80,18 +97,40 @@ export const MultipleSelect = ({
   };
 
   const handleSelect = (newValue: string) => {
-    const selectedUpdated =
-      value && value?.length > 0 ? [...value, newValue] : [newValue];
-
-    onChange(selectedUpdated);
+    const selectedUpdated = multipleSelected
+      ? multipleSelected && multipleSelected?.length > 0
+        ? [...multipleSelected, newValue]
+        : [newValue]
+      : value && value?.length > 0
+      ? [...value, newValue]
+      : [newValue];
+
+    if (onChangeMultipleSelected) {
+      onChangeMultipleSelected(selectedUpdated);
+    } else {
+      onChange(selectedUpdated);
+    }
   };
 
   const handleDeselect = (deselectValue: string) => {
```

Comment:
Uma forma muito mais simples seria:
-  um array de valores/labels gerais do select;
-  um array de valores selecionados;
-  uma única função de onChange que recebe o value clicado e busca o valor na array de selecionados. Se for multiselect - ao encontrar o valor selecionado, remove; ao não encontrar, adiciona. Se for single-select - checa se o valor atual é igual ao selecionado e, se não, substitui.

Essa estrutura serve para multi select e single select - vc só precisa ter a flag mesmo. Quanto mais complexa for a solução que vc encontrar para um problema, maior a chance de bugs e comportamentos indesejados. Melhor sempre procurar uma solução simples e com poucas linhas de códigos.

Reply:
Esses componentes já existiam antes, por isso tudo que eu fiz foi pra adapta-los sem que precisasse mexer muito nas coisas que já estão funcionando. Por isso parece que tem coisas que foram meio que coladas no código